plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/IOpenShiftConnection.java

```diff
@@ -22,7 +24,7 @@
  * @author jeff.cantrill
  *
  */
-public interface IOpenShiftConnection {
+public interface IOpenShiftConnection extends IConnection {
```

Comment:
Agreed, it should, but you have to consider the history; the v3 code base evolved from v2 which is a completely different model.  There was(is?) a desire for the explorer to support both and as such steps were taken to accommodate not disrupting the v2 code base.  This change is probably not that big of an issue